File: 无重复字符的最长子串.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        mp = Counter()
        left, ans = 0, 0
        for right, c in enumerate(s):
            mp[c] += 1
            while mp[c] > 1:
                mp[s[left]] -= 1
                left += 1
            ans = max(ans, right - left + 1)
        return ans

File: 无重复字符的最长子串.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        map<char, int> cnt;
        int ans = 0, left = 0, right = 0;
        while (right < s.size()) {
            ++cnt[s[right]];
            while (cnt[s[right]] > 1) {
                --cnt[s[left]];
                left++;
            }
            ans = max(ans, right - left + 1);
            ++right;
        }
        return ans;
    }
};
